libvast/src/view.cpp

```diff
@@ -19,16 +19,38 @@ pattern_view::pattern_view(const pattern& x) : pattern_{x.string()} {
   // nop
 }
 
+bool operator==(pattern_view x, pattern_view y) noexcept {
+  return x.pattern_ == y.pattern_;
+}
+
 address_view::address_view(const address& x) : data_{&x.data()} {
   // nop
 }
 
+bool operator==(address_view x, address_view y) noexcept {
+  return x.data_ == y.data_;
+}
+
 subnet_view::subnet_view(const subnet& x)
   : network_{x.network()},
     length_{x.length()} {
   // nop
 }
 
+bool operator==(subnet_view x, subnet_view y) noexcept {
+  return x.network_ == y.network_ && x.length_ == y.length_;
+}
+
+vector_view::iterator vector_view::begin() const {
+  // TODO: const_cast okay?
+  return {vector_view_ptr{const_cast<vector_view*>(this), true}, 0};
```

Comment:
@Neverlord Do you have a better idea than `const_cast` here? The issue is that `begin`/`end` increase the ref count, hence the need for a non-const pointer. At the same time, this is an implementation detail. The user has always a const view of the data.

Reply:
Why is the iterator participating in the lifetime of the view? Isn't it more idiomatic (and cheaper) to have a `const vector_view*` in the iterator instead?